imxrt-hal/src/iomuxc.rs

```diff
@@ -1,4 +1,34 @@
 //! IOMUX Controller
+//!
+//! ## Configure pin settings
```

Comment:
This seems out of place to me, the module level docs here probably should encompass more of what iomuxc does as a whole rather than the PinConfig stuff

Reply:
I can move into the `pin_config.rs` module-level docs, rather than the `iomxc.rs` docs.